### PR TITLE
Fix ctest project.

### PR DESCRIPTION
**Pull request by wo80**
@jzarl 

<del>By making the triangle-api project a shared library, commit d5cc0de20d12efcee0289694c480d807984c8462 probably broke ctest.</del> To make testing work, the shared library must be copied to the test target directory.

This works with Visual Studio. Before merging, I'd like to make sure that it also works with other compiler platforms.

### CODE REVIEW

**Comment by jzarl** [comment]
On Linux (probably it's the same for MacOS), there's no dll lookup in the local directory. Copying the shared library file to `build_dir/examples/triangle-test` therefore won't help you.
However, cmake sets the RPATH for the binaries in the build directory, so that the library is found in `build_dir/Triangle`. I guess the cleanest solution is to only copy the file on windows.

One minor pitfall remains, though: While testing I had the problem that the installed version of the library was used instead of the (newer) one in the build directory. That was probably caused by me setting the `LD_LIBRARY_PATH`, I guess.

So in summary:
 - Copying the .so file won't help, but won't hurt either.
 - In a clean environment, ctest works fine on Linux
 - People setting `LD_LIBRARY_PATH` should know what they're doing, so that's fine as well.

**Comment by wo80** [comment]
Thanks for your insights! :+1:

I will copy the file only on windows platforms.
